src/coreclr/vm/exceptionhandling.cpp

```diff
@@ -3920,32 +3921,39 @@ extern "C" CLR_BOOL QCALLTYPE SfiNext(StackFrameIterator* pThis, uint* uExCollid
     ExInfo* pExInfo = pThis->GetNextExInfo();
     bool isCollided = false;
 
-    MethodDesc *pMD = pThis->m_crawl.GetFunction();
+    bool doingFuncletUnwind = pThis->m_crawl.IsFunclet();
```

Comment:
Note: We no longer update the `doingFuncletUnwind` value in each iteration of the loop. Presumably that's okay but it's an algorithmic difference.

Reply:
It seems we might want to set it to `false` right before the while to match the original behavior (it would always get false if it looped in the original state, because the frame state would not be "frame function" and so it cannot be a funclet.